geode-core/src/main/java/org/apache/geode/distributed/internal/InternalDistributedSystem.java

```diff
@@ -2167,7 +2191,7 @@ private static void notifyReconnectListeners(InternalDistributedSystem oldsys,
   private void notifyResourceEventListeners(ResourceEvent event, Object resource) {
     for (Iterator<ResourceEventsListener> iter = resourceListeners.iterator(); iter.hasNext();) {
```

Comment:
This loop looks like it could be rewritten  as a foreach type loop

`
for (ResourceEventListnener listener:resourceListeners) {
  ...
}
`